### DISCUSSION TITLE
Some questions in ComputeSmallStrain.C

### DISCUSSION THREAD

**Discussion by Nuwanted** [idaholab/moose#26418] [Q&A Modules: Solid mechanics] [answered]
I found some code in the ComputeSmallStrain.C file about volumetric strain.  I would like to ask if the first term on the right side of the formula in the code is the total strain trace, and what do the second and third terms represent?
volumetric_strain += _total_strain[_qp].trace() * _JxW[_qp] * _coord[_qp];
What do (0,0), (1,1), and (2,2) mean in the following code?
 if (_volumetric_locking_correction)
    {
      Real trace = _total_strain[_qp].trace();
      _total_strain[_qp](0, 0) += (volumetric_strain - trace) / 3.0;
      _total_strain[_qp](1, 1) += (volumetric_strain - trace) / 3.0;
      _total_strain[_qp](2, 2) += (volumetric_strain - trace) / 3.0;
Thanks for your reply.

**Comment by GiudGiud** [comment, marked as answer]
Hello

```
 _JxW[_qp]
```
is the quadrature weights and the Jacobian for the transformation from the real element to the reference element
```
_coord[_qp]
```
is a factor for the integration in RZ/ RSpherical coordinates. In RZ the size of the infinitesimal integration quantity is 2pi r dr for example, so _coord is 2 pi r. (in xyz, domega = dxdyxdz so `_coord = 1`)

```
(0, 0) etc
```
are simply indices into the diagonal of the tensor. 

Guillaume
